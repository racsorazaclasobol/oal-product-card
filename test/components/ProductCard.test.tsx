import React from 'react';
import rendered, { act } from 'react-test-renderer';
import ProductCard, { ProductTitle } from '../../src/components';
import { product1 } from '../data/products'

describe('ProductTitle', () => {

	test('Debe de mostrar el componente', () => {

		const wrapper = rendered.create(
			<ProductCard product={ product1 }>
				{
					() => (
						<ProductTitle />
					)
				}
			</ProductCard>
		);

		expect(wrapper.toJSON()).toMatchSnapshot();
		
    });

    test('Debe de aumentar el contador', () => { 
        
        const wrapper = rendered.create(
			<ProductCard product={ product1 }>
				{
					({ counter, increaseBy }) => (
                        <>
                            <h1>Test Product</h1>
                            <span> { counter } </span>
                            <button onClick={ () => increaseBy( 1 ) } ></button>
                        </>					
					)
				}
			</ProductCard>
		);
        let wrapperJSON = wrapper.toJSON();
        expect(wrapper.toJSON()).toMatchSnapshot();

       act( () => { ( wrapperJSON as any ).children[2].props.onClick() } );
       
       wrapperJSON = wrapper.toJSON();
       
       expect( ( wrapperJSON as any ).children[1].children[1] ).toBe( '1' );

    })
});
