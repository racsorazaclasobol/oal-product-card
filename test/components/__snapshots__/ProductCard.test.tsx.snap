// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle Debe de aumentar el contador 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Test Product
  </h1>
  <span>
     
    0
     
  </span>
  <button
    onClick={[Function]}
  />
</div>
`;

exports[`ProductTitle Debe de mostrar el componente 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
     
    Coffee Mug - Sin Imagen
     
  </span>
</div>
`;
